api/allocation_api.py

```diff
@@ -0,0 +1,123 @@
+import threading
+
+from django.conf import settings
+from django.utils.datetime_safe import datetime
+from rest_framework import permissions, serializers, viewsets
+from rest_framework.exceptions import ValidationError
+
+from allocation.allocation_data_builder import AllocationDataBuilder
+from allocation.allocation_solver import AllocationSolver
+from allocation.models import AllocationRequest
+from api.applications_api.serializers import NullableCurrentUserDefault
+from applications.models import (
+    ApplicationRound,
+    ApplicationRoundBasket,
+    ApplicationRoundStatus,
+)
+from permissions.api_permissions import AllocationRequestPermission
+
+
+class AllocationInProcessException(Exception):
+    pass
+
+
+class UnsuitableStatusForAllocationException(Exception):
+    pass
+
+
+class AllocationRequestSerializer(serializers.ModelSerializer):
```

Comment:
So for @kieferbonk 

This is a way to start allocating specifiv allocation round (and selected baskets)

Basically at minimum you only need application_round_id, 

This can also track whether it has completed and when, maybe we should actually rename completed to something else, like succeeded? Basically currently if there is no end date it has not completed allocating, and the boolean completed means if false some exception happened (they shouldn't) if true allocation completed successfully. 

When starting the allocation it also marks application round's allocating to true and sets it back to false when completed (or failed)